packages/xod-client-electron/src/app/main.js

```diff
@@ -266,23 +268,42 @@ const onReady = () => {
       )
       .then(arduinoCli => {
         arduinoCliInstance = arduinoCli;
-        subscribeListBoards(arduinoCli);
-        subscribeUpload(arduinoCli);
-        subscribeUpdateIndexes(arduinoCli);
-        subscribeCheckUpdates(arduinoCli);
-        subscribeUpgradeArduinoPackages(arduinoCli);
+
+        // unsubscribe old listeners
+        unsubscribers.forEach(R.call);
 
         // On switching workspace -> update arduino-cli config and run migration
-        ipcMain.on(EVENTS.SWITCH_WORKSPACE, (event, newWsPath) =>
+        const onSwitchWorkspace = (event, newWsPath) =>
           xdb.switchWorkspace(
             arduinoCli,
             getPathToBundledWorkspace(),
             newWsPath
-          )
-        );
-
-        subscribeOnCheckArduinoDependencies(arduinoCli);
-        subscribeOnInstallArduinoDependencies(arduinoCli);
+          );
+        ipcMain.on(EVENTS.SWITCH_WORKSPACE, onSwitchWorkspace);
+
+        unsubscribers = [
+          () =>
+            ipcMain.removeListener(EVENTS.SWITCH_WORKSPACE, onSwitchWorkspace),
+          subscribeListBoards(arduinoCli),
+          subscribeUpload(arduinoCli),
+          subscribeUpdateIndexes(arduinoCli),
+          subscribeCheckUpdates(arduinoCli),
+          subscribeUpgradeArduinoPackages(arduinoCli),
+          subscribeOnCheckArduinoDependencies(arduinoCli),
+          subscribeOnInstallArduinoDependencies(arduinoCli),
+          store.subscribe(() => {
+            const projectPath = store.select.projectPath();
+            win.webContents.send(EVENTS.PROJECT_PATH_CHANGED, projectPath);
```

Comment:
After the refactoring, this block looks overcomplicated. Would you extract it to a separate function? `onProjectChange` or something

Reply:
The only thing stored in that store (at least for now) is `projectPath`. Do you think `onProjectPathChange` is a more appropriate name?

Reply:
Yet, `onProjectPathChange` is fine

Reply:
Okay! Done.